Event/FilterLocaleSwitchEvent.php

```diff
@@ -36,7 +36,7 @@ class FilterLocaleSwitchEvent extends Event
      */
     public function __construct(Request $request, $locale)
     {
-        if (!is_string($locale) || null == $locale || '' == $locale) {
+        if (!is_string($locale)) {
```

Comment:
because something is really wrong when the locale is an empty string. i would do `'' === $locale` - empty checks other things we don't care as null is not is_string.